src/lib/retry.go

```diff
@@ -0,0 +1,36 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lib
+
+import (
+	"math/rand"
+	"time"
+)
+
+// Retry retry max attempts of f unit it's successful
+func Retry(attempts int, sleep time.Duration, f func() error) error {
```

Comment:
1. This is opinionated but why not use a `bool` as the condition, when you wait for a condition it's not always  `err != nil`.  
2. Just curious, why you choose the exponential alg for the retry?  Normally it's for avoiding congestion.  We should also document this to avoid misuse.